src/coreclr/jit/gentree.cpp

```diff
@@ -24716,6 +24932,50 @@ GenTree* Compiler::gtNewSimdMaxNode(
     var_types simdBaseType = JitType2PreciseVarType(simdBaseJitType);
     assert(varTypeIsArithmetic(simdBaseType));
 
+#if defined(TARGET_XARCH)
+    if (varTypeIsFloating(simdBaseType))
+    {
+        GenTree* op1Dup1 = fgMakeMultiUse(&op1);
+        GenTree* op1Dup2 = fgMakeMultiUse(&op1);
+        GenTree* op1Dup3 = fgMakeMultiUse(&op1);
```

Comment:
is this a correct way to get 3 copies? `fgMakeMultiUse(&op1)` will modify `op1` to be COMMA (if it's complex) to do temp assignment and then you make that comma multi-use again. I think you need to call `gtCloneExpr` on top of resulting dup instead.

Reply:
Ah right, my mistake. Will fix

Reply:
Fixed this and a couple other places that were doing the same thing.